src/LinearAlgebra/SparseMatrix/COO_Matrix.hpp

```diff
@@ -189,14 +189,15 @@ inline std::vector<IdxT> COO_Matrix<ScalarT, IdxT>::getCSRRowData()
 {
   if (!this->isSorted())
     this->sortSparse();
-  std::vector<IdxT> row_size_vec(this->rows_size_ + 1, 0);
+  std::vector<IdxT> row_size_vec(static_cast<size_t>(this->rows_size_ + 1), 0);
   IdxT              counter = 0;
   for (IdxT i = 0; i < static_cast<IdxT>(row_size_vec.size() - 1); i++)
```

Comment:
I think as discussed in the original issue, these should just be changed to `size_t` rather than `static_cast`ing them everywhere they're used. `IdxT` should be used for sparse indices rather than for only indexing stl containers